Migrate openHABCore to Swift 6

timbms opened the pull request.

timbms committed:
Migrate openHABCore to Swift 6

Signed-off-by: Tim Mueller-Seydlitz <timbms@gmail.com>

timbms committed:
Merge branch 'develop' into migrateOpenHABCoreToSwift6

timbms committed:
Migrated the ServerCertificateManagerTests from XCTest to Swift Testing, made tests compatible with Swift 6
Added @Test(...) attributes with descriptive names, Removed test prefix from method names, Used #expect() instead of XCTAssert* macros
Used nonisolated(unsafe) for mutable properties in the mock delegate, Made the mock delegate conform to @unchecked Sendable

Signed-off-by: Tim Mueller-Seydlitz <timbms@gmail.com>

timbms commented:
@TAKeanice: Please have a look at this PR.

timbms committed:
Migrate to iOS 18.4 for simulator tests

Signed-off-by: Tim Mueller-Seydlitz <timbms@gmail.com>

timbms committed:
Set the Xcode version to 16.4

Signed-off-by: Tim Mueller-Seydlitz <timbms@gmail.com>

timbms committed:
Disable upcoming features like 'DynamicActorIsolation' that are already enabled as of Swift version 6

Signed-off-by: Tim Mueller-Seydlitz <timbms@gmail.com>

timbms commented on OpenHABCore/Tests/OpenHABCoreTests/ServerCertificateManagerTests.swift:
```diff
@@ -9,40 +9,27 @@
+final class MockServerCertificateDelegate: ServerCertificateManagerDelegate, @unchecked Sendable {
+    nonisolated(unsafe) var lastCall = ""
+    nonisolated(unsafe) var expectedResult: ServerCertificateManager.EvaluateResult = .permitOnce
+    nonisolated(unsafe) var acceptedChangedCalled = false
 
-    func evaluateServerTrust(summary: String?, forDomain domain: String?) async -> ServerCertificateManager.EvaluateResult {
+    nonisolated func evaluateServerTrust(summary: String?, forDomain domain: String?) async -> ServerCertificateManager.EvaluateResult {
         lastCall = "evaluateServerTrust"
         return expectedResult
     }
 
-    func evaluateCertificateMismatch(summary: String?, forDomain domain: String?) async -> ServerCertificateManager.EvaluateResult {
+    nonisolated func evaluateCertificateMismatch(summary: String?, forDomain domain: String?) async -> ServerCertificateManager.EvaluateResult {
         lastCall = "evaluateCertificateMismatch"
         return expectedResult
     }
 
-    func acceptedServerCertificatesChanged() {
+    nonisolated func acceptedServerCertificatesChanged() {
```
I know that I am bypassing concurrency safety. At least it unblocks the current situation for testing.

timbms committed:
Merge branch 'develop' into migrateOpenHABCoreToSwift6

Signed-off-by: Tim Bert <5411131+timbms@users.noreply.github.com>

TAKeanice commented:
> > @TAKeanice Hopefully you can find a workaround.
> 
> I will work on it tomorrow early morning!

That morning was spent trying to get XCode26 onto my system... I will try and solve the problem tomorrow, but I already got a pretty good idea about the problem from the description on the swiftlang discussion.

TAKeanice committed:
attempt to make Preferences thread-safe

Swift 6.2 throws concurrency checker errors on static properties in actors, since they are not isolated

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice committed:
preliminary concurrency "fixes" for intents and some views

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice committed:
More Swift 6.2 concurrency fixes

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice committed:
remove unnecessary "Task"s

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice committed:
More logging for debugging long wait times when changing homes

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice committed:
Isolate preferences to main actor

to avoid problems with calls from synchronous main-actor isolated code

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice commented:
My XCode 16 now compiles the code, and my phone seems to run everything properly.

timbms commented:
Still. You found a solution!
One more thing: You may have seen that the deployment to TestFlight failed. I don't understand why the final compile run after test failed.

timbms commented:
Have a look at https://github.com/openhab/openhab-ios/pull/968.
That fixed the issue.
